<?xml version="1.0" encoding="UTF-8"?>
<!-- SPDX-License-Identifier: BSD-3-Clause -->
<!-- SPDX-FileCopyrightText: Czech Technical University in Prague -->

<!-- Run magnetometer compass to provide azimuth. -->

<launch>
    <arg name="target_unit" default="deg" description="-" />
    <arg name="target_orientation" default="ned" description="-" />
    <arg name="target_reference" default="magnetic" description="-" />
    <arg name="target_type" default="azimuth" description="-" />
    <arg name="target_append_suffix" default="false" description="-" />
    <arg name="target_frame" default="base_link_no_grav" description="-" />
    <arg name="out_frame_id" default="" description="-" />
    <arg name="magnetic_declination" default="-0.068" description="Set magnetic declination to zero in simulation." />

    <arg name="input_orientation" default="ned" description="-" />
    <arg name="input_reference" default="magnetic" description="-" />

    <node pkg="compass_conversions" exec="compass_transformer_nodelet" name="compass" respawn="true">
        <param name="magnetic_declination" value="$(var magnetic_declination)" />
        <param name="target_unit" value="$(var target_unit)" />
        <param name="target_orientation" value="$(var target_orientation)" />
        <param name="target_reference" value="$(var target_reference)" />
        <param name="target_type" value="$(var target_type)" />
        <param name="target_append_suffix" value="$(var target_append_suffix)" />
        <param name="target_frame" value="$(var target_frame)" />
        <param name="out_frame_id" value="$(var out_frame_id)" />

        <param name="input_orientation" value="$(var input_orientation)" />
        <param name="input_reference" value="$(var input_reference)" />
        <remap from="/azimuth_in" to="/compass/mag/ned/deg" />
    </node>
</launch>